File: src/main/java/dev/tricht/lunaris/util/DirectoryManager.java
package dev.tricht.lunaris.util;

import java.io.File;

public class DirectoryManager {

    public static File getFile(String filename) {
        return new File(getDirectory().getAbsolutePath() + File.separator + filename);
    }

    public static File getDataDirectory(String folderName) {
        File dataDirectory = new File(getDirectory() + File.separator + "data"
                + (folderName.isEmpty() ? "" : File.separator + folderName));
        if (!dataDirectory.exists()) {
            if (!dataDirectory.mkdirs()) {
                throw new RuntimeException("Unable to create directory.");
            }
        }
        return dataDirectory;
    }

    private static File getDirectory() {
        File directory = null;
        if (Platform.isLinux()) {
            String home = System.getProperty("user.home");
            directory = new File(home + "/.config/Lunaris");
        } else if (Platform.isWindows()) {
            directory = new File(System.getenv("APPDATA") + "\\Lunaris");
        } else {
            ErrorUtil.showErrorDialogAndExit(String.format("Platform %s not supported.", Platform.getOS()));
        }
        if (!directory.exists()) {
            if (!directory.mkdirs()) {
                ErrorUtil.showErrorDialogAndExit("Unable to create lunaris data/config directory.");
            }
        }
        return directory;
    }

}

File: src/main/java/dev/tricht/lunaris/util/DirectoryManager.kt
package dev.tricht.lunaris.util

import java.io.File

object DirectoryManager {
    private val directory: File?
        get() {
            val directory = when {
                Platform.isLinux -> {
                    val home = System.getProperty("user.home")
                    File("$home/.config/Lunaris")
                }
                Platform.isWindows -> File(System.getenv("APPDATA") + "\\Lunaris")
                else -> null
            }
            if (directory == null) {
                ErrorUtil.showErrorDialogAndExit(String.format("Platform %s not supported.", Platform.os))
            }
            if (directory?.exists() == false) {
                if (!directory.mkdirs()) {
                    ErrorUtil.showErrorDialogAndExit("Unable to create lunaris data/config directory.")
                }
            }
            return directory
        }

    fun getFile(filename: String): File {
        return File(directory?.absolutePath + File.separator + filename)
    }

    fun getDataDirectory(folderName: String): File {
        val dataDirectory = File(directory?.toString() + File.separator + "data"
                + if (folderName.isEmpty()) "" else File.separator + folderName)
        if (!dataDirectory.exists()) {
            if (!dataDirectory.mkdirs()) {
                throw RuntimeException("Unable to create directory.")
            }
        }
        return dataDirectory
    }
}
